features/FEATURE_COMMON_PAL/mbed-coap/source/sn_coap_protocol.c

```diff
@@ -641,11 +639,21 @@ sn_coap_hdr_s *sn_coap_protocol_parse(struct coap_s *handle, sn_nsdl_addr_s *src
                 break;
             }
         }
-
+        /* Remove from the list if not an notification message.
+         * Initial notification message is needed for sending rest of the blocks (GET request).
+        */
+        bool remove_from_the_list = false;
         if (stored_blockwise_msg_temp_ptr) {
-            tr_debug("sn_coap_protocol_parse - remove block message %d", stored_blockwise_msg_temp_ptr->coap_msg_ptr->msg_id);
+            if (stored_blockwise_msg_temp_ptr->coap_msg_ptr &&
+                stored_blockwise_msg_temp_ptr->coap_msg_ptr->options_list_ptr &&
+                stored_blockwise_msg_temp_ptr->coap_msg_ptr->options_list_ptr->observe != COAP_OBSERVE_NONE) {
+                remove_from_the_list = false;
+            } else {
```

Comment:
If remove_from_list was initialised to true , the else block would be unnecessary?